File: lib/qt5/ItemListComboBox.cpp
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <pkmn/qt5/ItemListComboBox.hpp>

#include <pkmn/database/lists.hpp>

namespace pkmn { namespace qt5 {

    ItemListComboBox::ItemListComboBox(
        const QString &game,
        QWidget* parent
    ): QComboBox(parent)
    {
        std::vector<std::string> items = pkmn::database::get_item_list(
                                             game.toStdString()
                                         );
        for(auto iter = items.begin(); iter != items.end(); ++iter) {
            addItem(QString::fromStdString(*iter));
        }

        setEditable(false);
    }

}}

// Generated at build time
#include "../include/pkmn/qt5/moc_ItemListComboBox.cpp"

File: lib/swig/java/ComboBoxes/ItemListComboBox.java
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.JComboBox;

/**
 * A JComboBox filled with a list of items in the given game.
 */
public class ItemListComboBox extends JComboBox<String> {
    public ItemListComboBox(String game) {
        StringVector itemList = Database.getItemList(game);
        for(long i = 0; i < itemList.size(); i++) {
            this.addItem(itemList.get((int)i));
        }
        this.setEditable(false);
    }
}
